teps/0091-trusted-resources.md

```diff
@@ -239,6 +263,64 @@ The signed Resource can be installed directly on the kubernetes cluster or built
 The verification should be done in the Tekton Pipeline's admission webhook. A new configmap should be added to gate the verification
 code.
 
+Before the verification, the signature is extracted from the resource. For the public key there are several options we can choose to config.
+
+Option 1: Configure the keys in configmap. The configmap cannot support a list of keys but one solution is that we can allow multiple keys be stored in one string and when fetching the keys we can parse it to get the list of keys. The string may follow [JSONPath](https://tools.ietf.org/id/draft-goessner-dispatch-jsonpath-00.html)  syntax of array. The con is that simple typo may cause parsing errors.
+
+Option 2: Add a new field `keyref` to refer to the key path
+```yaml
+kind: TaskRun
+apiVersion: tekton.dev/v1beta1
+metadata:
+ name: example-tr
+spec:
+ taskRef:
+   name: example-task
+   keyref: keypath # this defines the public key of this taskref
+```
+
+```yaml
+kind: PipelineRun
+apiVersion: tekton.dev/v1beta1
+metadata:
+ name: example-pr
+spec:
+ pipelineRef:
+   name: example-pipeline
+   keyref: keypath # this defines the public key of this pipelineref
+```
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+ annotations:
+   tekton.dev/signature: MEQCIHhFC480mv7mh/6dxAlp/mvuvGXanuSghMsT+iBhWxt5AiBDvHv8sfKjJ3Ozrzvp+cjr30AOx1SPQDCcaJjlpAoVrA==
+ name: example-Pipeline
+spec:
+ tasks:
+   - name: example-task
+     taskRef:
+       name: example-task1
+       keyref: keypath1  # this defines the public key of this taskref
+   - name: example-task2
+     taskRef:
+       name: example-task2
+       keyref: keypath2  # this defines the public key of this taskref
+```
+`keyref` defines the path of public key of this taskRef. All the tasks within a pipeline need to be signed off, and they may come from different sources with different public keys.
+So a `keyref` is needed when signing the pipeline. The signing cli doesn't need to verify the referred keys when signing pipeline.
+
+**Pros vs Cons:**
+| Option | Pros | Cons |
+| -------- | ----------- | ----------- |
+| Config at install level (e.g. configmap) | Easy to config at a central place |  ConfigMap doesn't support list, may need to parse the string or create a new cluster level config to store the keys
+| Add keyref in taskRef&pipelineRef | Each resource can speficy the corresponding key, no need to loop all available keys for one resource's verification| Introduce cost for users to write and maintain YAML files
```

Comment:
The keyref is not going together with signature so it may not be an issue?
For example, is keyref is configured at taskrun and signature is at task. The attacker doesn't access to taskrun